### PR TITLE
[QA failed: do not check this off] fix: Submit expense option is not disabled for users who are invited to expense report

### PR DESCRIPTION
**Pull request by nkdengineer**
### Fixed Issues

$ https://github.com/Expensify/App/issues/46340
PROPOSAL: https://github.com/Expensify/App/issues/46340#issuecomment-2253790217

### Tests

1. [User A] Submit two expenses to User B
2. [User A] Click on the expense preview in the main chat to go to expense report
3. [User A] Send a user mention containing User C
4. [User A] Click Invite from the whisper message
5. [User A] Send a message in the transaction thread
6. [User C] Open the invited expense report
9. Verify that: For IOUs, only two users can add expenses to the report, the manager (report.managerID) and the submitter (report.accountID).
10. Verify that: For expense reports, only the submitter/owner should be able to add expenses.

- [x] Verify that no errors appear in the JS console

### QA Steps

1. [User A] Submit two expenses to User B
2. [User A] Click on the expense preview in the main chat to go to expense report
3. [User A] Send a user mention containing User C
4. [User A] Click Invite from the whisper message
5. [User A] Send a message in the transaction thread
6. [User C] Open the invited expense report
9. Verify that: For IOUs, only two users can add expenses to the report, the manager (report.managerID) and the submitter (report.accountID).
10. Verify that: For expense reports, only the submitter/owner should be able to add expenses.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/5248b737-4096-4446-8c2a-e73fb8bfc2d0

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/2a7d3f0d-7f0c-44ed-9dd5-10df1ee44b23

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/b8e7aa15-2b4e-48d2-bd8a-46cb47f58524

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/4f24aeba-2391-41a7-93be-cc3df2261ff5

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/9577d910-235e-4e2e-af65-53b772516be6

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/a8be656f-ff9c-4ff1-8ac0-31df919699e1

</details>

### CODE REVIEW

File: src/libs/ReportUtils.ts [hunk 1]
```diff
@@ -6218,7 +6226,7 @@ function canRequestMoney(report: OnyxEntry<Report>, policy: OnyxEntry<Policy>, o
     // which is tied to their workspace chat.
     if (isMoneyRequestReport(report)) {
         const canAddTransactions = canAddTransaction(report);
-        return isReportInGroupPolicy(report) ? isOwnPolicyExpenseChat && canAddTransactions : canAddTransactions;
+        return isReportInGroupPolicy(report) || isExpenseReport(report) ? isOwnPolicyExpenseChat && canAddTransactions : canAddTransactions;
```
[the remark this answers is not in the document]
Comment (nkdengineer) [reply src/libs/ReportUtils.ts]: @dukenv0307 I updated

**Review by dukenv0307 — APPROVED** [review]

**Review by aldo-expensify — APPROVED** [review]

File: src/libs/ReportUtils.ts [hunk 2]
```diff
@@ -1660,6 +1660,14 @@ function canAddOrDeleteTransactions(moneyRequestReport: OnyxEntry<Report>): bool
         return false;
     }
 
+    if (isIOUReport(moneyRequestReport) && currentUserAccountID !== moneyRequestReport?.managerID && currentUserAccountID !== moneyRequestReport?.ownerAccountID) {
+        return false;
+    }
+
+    if (isExpenseReport(moneyRequestReport) && currentUserAccountID !== moneyRequestReport?.managerID) {
```
Comment (youssef-lr) [inline src/libs/ReportUtils.ts:1667]: Wouldn't this break Instant Submit report where submitters create an expense and it gets added to a processing report?

Comment (youssef-lr) [reply src/libs/ReportUtils.ts:1667]: Yeah this breaks optimistic data for instant submit, we end up creating a new stale preview, but the backend adds the expense to the processing report


Uploading Screen Recording 2024-08-16 at 22.24.26.mov…

**Comment by youssef-lr** [comment]
I think we need to do the check in this function https://github.com/Expensify/App/blob/0214b3fb311a64c5280576fd352327ac6f9fe5c9/src/libs/ReportUtils.ts#L6264

**Comment by aldo-expensify** [comment]
@youssef-lr since this has not been deployed yet, what about just creating a revert PR and let @nkdengineer do a new PR that works well without needing to rush because of a deploy blocker. Sounds good?

**Comment by trjExpensify** [comment]
> AFAIK only submitters can see that button, but I'll let @trjExpensify or @JmillsExpensify confirm. Who can add expenses to a submitted report, the submitter only, or submitter, manager, and admin?

Yep, only the submitter can add expenses to the expense report.

**Comment by youssef-lr** [comment]
Sounds good @aldo-expensify

**Comment by nkdengineer** [comment]
I'll create follow PR soon

**Comment by nkdengineer** [comment]
@youssef-lr I don't know how to reproduce [this bug](https://github.com/Expensify/App/pull/47373#discussion_r1720369904)
Please write in detail the steps and explain the problem we are facing 🙏
Thank you

**Comment by youssef-lr** [comment]
1. Create a workspace and add a member
2. As the member, submit an expense from the workspace chat
3. Open the expense
4. Click '+' and notice you can't see 'Submit expense'

**Comment by kavimuru** [comment]
@youssef-lr are we good to QA this PR?

**Comment by nkdengineer** [comment]
@kavimuru This PR was reverted so we don't need to QA this now.

**Comment by aldo-expensify** [comment]
Thanks for handling the revert @youssef-lr

**Pull request merged**